examples/GoPlatform/platform/main.go

```diff
@@ -0,0 +1,60 @@
+package main
+
+/*
+#cgo LDFLAGS: ./main.o -ldl
+#include "./host.h"
+*/
+import "C"
+
+import (
+	"fmt"
+	"unsafe"
+)
+
+func main() {
+	var str C.struct_RocStr
+	C.roc__mainForHost_1_exposed_generic(&str)
+	fmt.Print(rocStrRead(str))
+}
+
+const is64Bit = uint64(^uintptr(0)) == ^uint64(0)
+
+func rocStrRead(rocStr C.struct_RocStr) string {
+	if int(rocStr.capacity) < 0 {
+		// Small string
+		ptr := (*byte)(unsafe.Pointer(&rocStr))
+
+		byteLen := 12
+		if is64Bit {
+			byteLen = 24
+		}
+
+		shortStr := unsafe.String(ptr, byteLen)
+		len := shortStr[byteLen-1] ^ 128
+		return shortStr[:len]
+	}
+
+	// Remove the bit for seamless string
+	seamlessBit := 31
+	if is64Bit {
+		seamlessBit = 63
+	}
+	len := uint64(rocStr.len) & ^uint64(1<<seamlessBit)
```

Comment:
Nit, but I think this could just be something like this:
```suggestion
	len := (uint(rocStr.len) << 1) >> 1
```